File: Java/src/test/java/com/interviewkata/InterviewKataTest.java
package com.interviewkata;

import jdk.nashorn.internal.ir.annotations.Ignore;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class InterviewKataTest {

    @Test
    void test_chico_sin_regalo_deseado_recibe_un_chocolate() {
        Santa santa = new Santa();
        Child jorge = new Child();
        jorge.setName("Jorge");

        santa.deliverGift(jorge);

        assertEquals("Soy Jorge y recibi un Chocolate", jorge.showMeWhatYouGot());
    }

    @Test
    void test_santa_sin_regalo_deseado_entrega_un_chocolate() {
        Santa santa = new Santa();
        Child laura = new Child();
        laura.setName("Laura");
        laura.setDesiredGift(new Gift("Reno"));

        santa.deliverGift(laura);

        assertEquals("Soy Laura y recibi un Chocolate", laura.showMeWhatYouGot());
    }

    @Test
    @Ignore // Este test este test falla
    void test_santa_con_regalo_deseado_entrega_el_regalo_correspondiente() {
        Santa santa = new Santa();
        Child juan = new Child();
        juan.setName("Juan");
        juan.setDesiredGift(new Gift("Trencito"));

        santa.deliverGift(juan);

        assertEquals("Soy Juan y recibi un Trencito", juan.showMeWhatYouGot());
    }
}

File: Kotlin/src/test/java/com/interviewkata/InterviewKataTest.kt
package com.interviewkata

import jdk.nashorn.internal.ir.annotations.Ignore
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class InterviewKataTest {
    @Test
    fun `chico sin regalo deseado, recibe un chocolate`() {
        val santa = Santa()
        val jorge = Child()
        jorge.name = "Jorge"

        santa.deliverGift(jorge)

        Assertions.assertEquals("Soy Jorge y recibi un Chocolate", jorge.showMeWhatYouGot())
    }

    @Test
    fun `santa sin regalo deseado en su bolsa, entrega un chocolate`() {
        val santa = Santa()
        val laura = Child()
        laura.name = "Laura"
        laura.desiredGift = Gift("Reno")

        santa.deliverGift(laura)

        Assertions.assertEquals("Soy Laura y recibi un Chocolate", laura.showMeWhatYouGot())
    }

    @Test
    @Ignore // Este test este test falla
    fun `santa con regalo deseado en su bolsa, entrega el regalo correspondiente`() {
        val santa = Santa()
        val juan = Child()
        juan.name = "Juan"
        juan.desiredGift = Gift("Trencito")

        santa.deliverGift(juan)

        Assertions.assertEquals("Soy Juan y recibi un Trencito", juan.showMeWhatYouGot())
    }
}
